code/datums/antagonists/datum_vampire.dm

```diff
@@ -0,0 +1,280 @@
+		L.adjustFireLoss(3)
+	else if(L.health < 50)
+		if(!L.on_fire)
+			L.visible_message("<span class='warning'>[L] catches fire!</span>", "<span class='danger'>Your skin catches fire!</span>")
+			L.emote("scream")
+		else
+			L.visible_message("<span class='warning'>[L] continues to burn!</span>", "<span class='danger'>Your continue to burn!</span>")
+		L.adjust_fire_stacks(5)
+		L.IgniteMob()
+	return
+
+/datum/antagonist/vampire/proc/check_sun()
+	var/mob/living/carbon/C = owner.current
+	if(!C)
+		return
+	var/ax = C.x
+	var/ay = C.y
+
+	for(var/i = 1 to 20)
+		ax += SSsun.dx
+		ay += SSsun.dy
+
+		var/turf/T = locate(round(ax, 0.5), round(ay, 0.5), C.z)
+
+		if(T.x == 1 || T.x == world.maxx || T.y == 1 || T.y == world.maxy)
+			break
+
+		if(T.density)
+			return
+	vamp_burn(TRUE)
+
+/datum/antagonist/vampire/proc/vampire_life()
+	var/mob/living/carbon/C = owner.current
+	if(!C)
+		return
+	if(owner && C.hud_used && C.hud_used.vamp_blood_display)
+		C.hud_used.vamp_blood_display.invisibility = FALSE
+		C.hud_used.vamp_blood_display.maptext = "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='#dd66dd'>[round(usable_blood, 1)]</font></div>"
+	handle_vampire_cloak()
+	if(istype(C.loc, /obj/structure/closet/coffin))
+		C.adjustBruteLoss(-4)
+		C.adjustFireLoss(-4)
+		C.adjustToxLoss(-4)
+		C.adjustOxyLoss(-4)
+		return
+	if(!get_ability(/datum/vampire_passive/full) && istype(get_area(C.loc), /area/chapel))
+		vamp_burn()
+	if(isspaceturf(C.loc))
+		check_sun()
+
+
+/datum/antagonist/vampire/proc/handle_bloodsucking(mob/living/carbon/human/H)
+	draining = H
+	var/mob/living/carbon/human/O = owner.current
+	var/blood = 0
+	var/old_bloodtotal = 0 //used to see if we increased our blood total
+	var/old_bloodusable = 0 //used to see if we increased our blood usable
+	log_attack("[O] ([O.ckey]) bit [H] ([H.ckey]) in the neck")
+	O.visible_message("<span class='danger'>[O] grabs [H]'s neck harshly and sinks in their fangs!</span>", "<span class='danger'>You sink your fangs into [H] and begin to drain their blood.</span>", "<span class='notice'>You hear a soft puncture and a wet sucking noise.</span>")
+	if(!iscarbon(owner))
+		H.LAssailant = null
+	else
+		H.LAssailant = O
+	playsound(O.loc, 'sound/weapons/bite.ogg', 50, 1)
+	while(do_mob(O, H, 50))
+		if(!is_vampire(O))
+			to_chat(O, "<span class='warning'>Your fangs have disappeared!</span>")
+			return
+		old_bloodtotal = total_blood
+		old_bloodusable = usable_blood
+		if(!H.blood_volume)
+			to_chat(O, "<span class='warning'>They've got no blood left to give.</span>")
+			break
+		if(H.stat != DEAD)
+			blood = min(20, H.blood_volume)// if they have less than 20 blood, give them the remnant else they get 20 blood
+			total_blood += blood / 2	//divide by 2 to counted the double suction since removing cloneloss -Melandor0
+			usable_blood += blood / 2
+		else
+			blood = min(5, H.blood_volume)	// The dead only give 5 blood

```

Comment:
wasn't it living only?

Reply:
thats usable_blood look two lines up and two lines down